### DISCUSSION TITLE
Differences between Full End Device and Minimal End Device

### DISCUSSION THREAD

**Discussion by ferre111** [openthread/openthread#8098] [Q&A] [answered]
Hi!

I have one basic question about end devices. What is the difference between full and minimal end devices apart from multicast address-related differences? Are there any constrains for MED?
I found out form the `Thread Primer` that MED forwards all messages to its parent. Does FED do anything else with its messages?

BR,
Karol

**Comment by jwhui** [comment, marked as answer]
A Minimal Thread Device (MTD) delegates a number of functions to its parent (which is an FTD) to reduce memory and energy resources on the MTD. The parent is responsible for determining the proper path to forward outgoing messages. The FTD is also responsible for buffering messages if the MTD is a sleepy, so that the MTD can retrieve the buffered messages when it is most convenient.

The [Thread Primer - Node Roles and Types](https://openthread.io/guides/thread-primer/node-roles-and-types) page provides more detail.
